### ISSUE TITLE
Can't include foreign relation on pydantic_model_creator

### ISSUE THREAD

**Issue by Master-Y0da** [tortoise/tortoise-orm#772]
I have the following pydantic schema:

      # Paciente has several fields but I only want this two
      paciente_out_all = pydantic_model_creator(Paciente,  include=('id', 'departments'))`

**departments** is a foreign key relation to another table...and when I try this:

`return await paciente_out_all.from_queryset(Paciente.all())`

throws this error:

      in field_map_update
      del field_map[raw_field]
      KeyError: 'departments_id'

Any idea ?. Thanks for your help.

**Comment by marcoaaguiar** [comment]
I believe that the issue is that the models are not being initiated before you create the pydantic models:
https://tortoise-orm.readthedocs.io/en/latest/contrib/pydantic.html#relations-early-init

**Comment by Master-Y0da** [comment]
@marcoaaguiar No is not. I already init the models with **Tortoise.init_models()**

**Comment by croixnet** [comment]
Hiya @Master-Y0da , @marcoaaguiar ,

Hmmmm ... would be interested to see your model implementations.  I believe I have a similar relationship and tried your limited include on the Pydantic model ...
```

from tortoise import fields, models, Tortoise, run_async
from tortoise.contrib.pydantic import pydantic_model_creator
import gzip


class Users(models.Model):
    """
    The User model
    """
    id = fields.IntField(pk=True)
    #: This is a username
    username = fields.CharField(max_length=20, unique=True)
    name = fields.CharField(max_length=50, null=True)
    family_name = fields.CharField(max_length=50, null=True)
    category = fields.CharField(max_length=30, default="misc")
    password_hash = fields.CharField(max_length=128, null=True)
    created_at = fields.DatetimeField(auto_now_add=True)
    modified_at = fields.DatetimeField(auto_now=True)

    chirps: fields.ReverseRelation['Chirp']

    class PydanticMeta:
        exclude = ["password_hash"]


class Chirp(models.Model):
    value = fields.TextField()
    user = fields.ForeignKeyField('models.Users', related_name='chirps', description='The user who owns this chirp')


Tortoise.init_models(['new_models'], 'models')
User_Pydantic = pydantic_model_creator(Users, name="User", include=('name', 'chirps'))  # <<< ------ Similar to your model
UserIn_Pydantic = pydantic_model_creator(Users, name="UserIn", exclude_readonly=True)
Chirp_Pydantic = pydantic_model_creator(Chirp, name="Chirp", include=('id', 'value',))
ChirpIn_Pydantic = pydantic_model_creator(Chirp, name="ChirpIn", exclude_readonly=True)


```
In my app entry point ...

```
from fastapi import FastAPI
from tortoise.contrib.fastapi import HTTPNotFoundError, register_tortoise
from new_models import Users, Chirp
from new_models import User_Pydantic, UserIn_Pydantic,  Chirp_Pydantic,  ChirpIn_Pydantic
from typing import List
from tortoise import Tortoise
import json
from pydantic import create_model

app = FastAPI(title="Tortoise ORM FastAPI example")

register_tortoise(
    app,
    db_url="postgres://iron:xxxxxxxx@127.0.0.1:5432/iron",
    modules={"models": ['new_models']},
    generate_schemas=True,
    add_exception_handlers=True,
)


@app.post("/users", response_model=User_Pydantic)
async def create_user(user: UserIn_Pydantic):
    user_obj = await Users.create(**user.dict(exclude_unset=True))
    return await User_Pydantic.from_tortoise_orm(user_obj)


@app.get("/user", response_model=List[User_Pydantic], responses={404: {"model": HTTPNotFoundError}})
async def get_users():
    return await User_Pydantic.from_queryset(Users.all())  # <<< ------- Similar to your call?


@app.get("/chirps/{user_id}", response_model=List[Chirp_Pydantic], responses={404: {"model": HTTPNotFoundError}})
async def get_chirps(user_id: int):
    return await Chirp_Pydantic.from_queryset(Chirp.filter(user_id=user_id))


@app.post("/chirps", response_model=Chirp_Pydantic)
async def create_chirp(chirp: ChirpIn_Pydantic):
    chirp_obj = await Chirp.create(**chirp.dict(exclude_unset=True))
    return await Chirp_Pydantic.from_tortoise_orm(chirp_obj)


```
I have a similar relationship between my Users model and the get_users() method.  Hope it all makes sense :-)

Cheers

**Comment by marcoaaguiar** [comment]
@Master-Y0da If you have more than one pydantic model from the same tortoise you have to pass a name as an argument, otherwise it will return the first PydanticModel created. It is a known issue: #647, #594. It is fixed by #735, but it is still waiting to be merged

**Comment by Master-Y0da** [comment]
@croixnet Our model implementation is very similar...only difference is that I don't use this line:

`chirps: fields.ReverseRelation['Chirp']`

I only declare the foreign key relation as you do in chirp model.

@marcoaaguiar I already did that, my pydantic models are named.

**Comment by marcoaaguiar** [comment]
In the snippet that you sent, they are not named. It should be something like this

```python
paciente_out_all = pydantic_model_creator(Paciente,  name="PacienteOutAll", include=('id', 'departments'))`
```

If you do `paciente_out_all.schema_json(indent=2)`, does it contain the departments field and respective model?

**Comment by croixnet** [comment]
Thanks @marcoaaguiar for the note on #647 and #594!

**Comment by Master-Y0da** [comment]
@marcoaaguiar That's exactly how I declare the pydantic model...and yes json schema shows departments field

**Comment by croixnet** [comment]
Hi @Master-Y0da , any chance we could see your models definition?

**Comment by croixnet** [comment]
Hello @Master-Y0da , @marcoaaguiar , I'm able to recreate the problem using these relationships ...

```

class Users(models.Model):
    id = fields.IntField(pk=True)
    name = fields.CharField(max_length=50, null=True)


class Klingon(models.Model):
    isklingon = fields.BooleanField()


class Cylon(models.Model):
    iscylon = fields.BooleanField()
    user = fields.ForeignKeyField('models.Users', related_name='cylon')
    user = fields.ForeignKeyField('models.Klingon', related_name='cylon')  # <=== duplicate 'user' variable

```
Any chance such a condition exists for you?

Cheers

**Comment by croixnet** [comment]
P.S. @Master-Y0da @marcoaaguiar  ... more to the story ...

Given a simple 'Users' model ...

```
class Users(models.Model):
    id = fields.IntField(pk=True)
    username = fields.CharField(max_length=20, unique=True)
    name = fields.CharField(max_length=50, null=True)
```

The following fails ... (per marcoaaguiar's key error) ...

```
class Cylon(models.Model):
    iscylon = fields.BooleanField()
    user = fields.ForeignKeyField('models.Users', related_name='cylon')  # <<=== user variable

Tortoise.init_models(['new_models'], 'models')
Cylon_Pydantic = pydantic_model_creator(Cylon, name="Cylon", include=('iscylon', 'user'))  # <<=== user reference

```

The following works ...

```
class Cylon(models.Model):
    iscylon = fields.BooleanField()
    user = fields.ForeignKeyField('models.Users', related_name='cylon')  # <<== creates user_id

Tortoise.init_models(['new_models'], 'models')
Cylon_Pydantic = pydantic_model_creator(Cylon, name="Cylon", include=('iscylon', 'user_id'))  # <<== reference user_id not user
```
Sadly you only get the user_id, not the user.  Still looking at solution.

Cheers

**Comment by Master-Y0da** [comment]
When I declare a foreig key...in this case **departments** ...tortoise adds '_id' at creating the schemas in database. So when I create a pydantic model for **paciente** I think it can't match **departments** with **departments_id** ...and that's the error. Sadly I can't find a solution.

**Comment by Abdeldjalil-H** [comment]
This is fixed by #1683. It should work on new versions.
